# Teste 1
#### Valor : Menção extra

### Faça os exercícios considerando a seguinte lista:

In [41]:
lista = [2, 29, 8, 98, 10, 3, -4, 9, 10]

#### Usando as funções apresentadas, faça de maneira dinâmica os seguintes exercícios:

1) Imprima a soma do maior e o menor elemento da lista:

In [4]:
menor = min(lista)

In [6]:
maior = max(lista)

In [15]:
somaMaiorMenor = lambda menor, maior:menor + maior

In [17]:
somaMaiorMenor(menor, maior)

94

2) Imprima o número de ocorrências do maior elemento da lista:

In [19]:
lista.count(maior)

1

3) Imprima o elemento no meio da lista:

In [75]:
lista[4:5]

[10]

In [1]:
string = 'gninrael'

ModuleNotFoundError: No module named 'stack'

In [6]:
newstring = ''

In [7]:
for char in string:
    newstring.push(char)

AttributeError: 'str' object has no attribute 'push'

4) Imprima o maior elemento da lista elevado ao quadrado:

In [24]:
import math

In [25]:
math.pow(maior, 2)

9604.0

5) Crie uma lista com todos os números pares de 0 até 1000.

In [46]:
lista = [i for i in range(0, 1001)]

In [47]:
lista

[0,
 1,
 2,
 3,
 4,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 20,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28,
 29,
 30,
 31,
 32,
 33,
 34,
 35,
 36,
 37,
 38,
 39,
 40,
 41,
 42,
 43,
 44,
 45,
 46,
 47,
 48,
 49,
 50,
 51,
 52,
 53,
 54,
 55,
 56,
 57,
 58,
 59,
 60,
 61,
 62,
 63,
 64,
 65,
 66,
 67,
 68,
 69,
 70,
 71,
 72,
 73,
 74,
 75,
 76,
 77,
 78,
 79,
 80,
 81,
 82,
 83,
 84,
 85,
 86,
 87,
 88,
 89,
 90,
 91,
 92,
 93,
 94,
 95,
 96,
 97,
 98,
 99,
 100,
 101,
 102,
 103,
 104,
 105,
 106,
 107,
 108,
 109,
 110,
 111,
 112,
 113,
 114,
 115,
 116,
 117,
 118,
 119,
 120,
 121,
 122,
 123,
 124,
 125,
 126,
 127,
 128,
 129,
 130,
 131,
 132,
 133,
 134,
 135,
 136,
 137,
 138,
 139,
 140,
 141,
 142,
 143,
 144,
 145,
 146,
 147,
 148,
 149,
 150,
 151,
 152,
 153,
 154,
 155,
 156,
 157,
 158,
 159,
 160,
 161,
 162,
 163,
 164,
 165,
 166,
 167,
 168,
 169,
 170,
 171,
 172,
 173,
 174,
 175,
 176,
 177,
 178,
 179,
 180,
 181,
 182,
 183,
 184,


In [56]:
pares = [i for i in lista if i % 2 == 0]

In [57]:
print(pares)

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200, 202, 204, 206, 208, 210, 212, 214, 216, 218, 220, 222, 224, 226, 228, 230, 232, 234, 236, 238, 240, 242, 244, 246, 248, 250, 252, 254, 256, 258, 260, 262, 264, 266, 268, 270, 272, 274, 276, 278, 280, 282, 284, 286, 288, 290, 292, 294, 296, 298, 300, 302, 304, 306, 308, 310, 312, 314, 316, 318, 320, 322, 324, 326, 328, 330, 332, 334, 336, 338, 340, 342, 344, 346, 348, 350, 352, 354, 356, 358, 360, 362, 364, 366, 368, 370, 372, 374, 376, 378, 380, 382, 384, 386, 388, 390, 392, 394, 396, 398, 400, 402, 404, 406, 408, 410, 412, 414, 416, 418, 420,

6) Crie uma lista com todos os números ímpares de 0 até 10.000.

In [2]:
newLista = [i for i in range(0, 10001) if i % 2 != 0]

In [3]:
print(newLista)

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69, 71, 73, 75, 77, 79, 81, 83, 85, 87, 89, 91, 93, 95, 97, 99, 101, 103, 105, 107, 109, 111, 113, 115, 117, 119, 121, 123, 125, 127, 129, 131, 133, 135, 137, 139, 141, 143, 145, 147, 149, 151, 153, 155, 157, 159, 161, 163, 165, 167, 169, 171, 173, 175, 177, 179, 181, 183, 185, 187, 189, 191, 193, 195, 197, 199, 201, 203, 205, 207, 209, 211, 213, 215, 217, 219, 221, 223, 225, 227, 229, 231, 233, 235, 237, 239, 241, 243, 245, 247, 249, 251, 253, 255, 257, 259, 261, 263, 265, 267, 269, 271, 273, 275, 277, 279, 281, 283, 285, 287, 289, 291, 293, 295, 297, 299, 301, 303, 305, 307, 309, 311, 313, 315, 317, 319, 321, 323, 325, 327, 329, 331, 333, 335, 337, 339, 341, 343, 345, 347, 349, 351, 353, 355, 357, 359, 361, 363, 365, 367, 369, 371, 373, 375, 377, 379, 381, 383, 385, 387, 389, 391, 393, 395, 397, 399, 401, 403, 405, 407, 409, 411, 413, 415, 417, 419, 421,

7) Imprima o valor da média de todos os elementos da lista.

In [4]:
print(sum(newLista) / len(newLista))

5000.0


8) Crie 3 tuplas para representar as informações de alunos em uma disciplina, cada tupla deverá conter o nome do aluno seguido de suas duas notas.

In [5]:
tuplas = ("Marcos", 80, 90), ("Vitin", 60, 70), ("Nick", 50, 40)

9) Imprima a soma das notas de cada aluno do exercício 8, use as variáveis criadas por você:

In [15]:
for x in tuplas:
    notas = x[1] + x[2]
    print(notas)

170
130
90


10) Imprima o nome e a maior nota de cada aluno do exercício 8, use as variáveis criadas por você:

In [96]:
print('Nome:',tuplas[0][0], '- Maior nota:', [max(tuplas[0][1:])])
print('Nome:',tuplas[1][0], '- Maior nota:', [max(tuplas[1][1:])])
print('Nome:',tuplas[2][0], '- Maior nota:', [max(tuplas[2][1:])])

Nome: Marcos - Maior nota: [90]
Nome: Vitin - Maior nota: [70]
Nome: Nick - Maior nota: [50]


11) Construa dicionários para armazenar as notas de 5 alunos. Imprima o valor da maior nota.

In [30]:
obj:dict = {
    "Marcos": 100,
    "Vitin": 80,
    "Nicholas": 40,
    "Lucas": 95,
    "Renata": 79
}

In [31]:
print(obj)

{'Marcos': 100, 'Vitin': 80, 'Nicholas': 40, 'Lucas': 95, 'Renata': 79}


12) Em uma competição de salto em distância cada atleta tem direito a cinco saltos. O resultado do atleta será determinado pela média dos cinco valores restantes. Você deve fazer um programa que receba o nome e as cinco distâncias alcançadas pelo atleta em seus saltos e depois informe o nome, os saltos e a média dos saltos. O programa deve ser encerrado quando não for informado o nome do atleta. A saída do programa deve ser conforme o exemplo abaixo:

Atleta: Rodrigo
 
Primeiro Salto: 6.5 m

Segundo Salto: 6.1 m

Terceiro Salto: 6.2 m

Quarto Salto: 5.4 m

Quinto Salto: 5.3 m

Resultado final:

Atleta: Rodrigo

Saltos: 6.5 - 6.1 - 6.2 - 5.4 - 5.3

Média dos saltos: 5.9 m

In [74]:
def salto(nome, salto):
    print('Atleta: ' + nome)
    print('Primeiro Salto:', salto[0], 'm')
    print('Segundo Salto:', salto[1], 'm')
    print('Terceiro Salto:', salto[2], 'm')
    print('Quarto Salto:', salto[3], 'm')
    print('Quinto Salto:', salto[4], 'm')
    print('Resultado final:')
    print('Atleta: ' + nome)
    print('Saltos: {}'.format(str(salto[0]) + " - " + str(salto[1]) + " - " + str(salto[2])  + " - " +  str(salto[3])  + " - " +  str(salto[4])))
    soma = sum(salto)
    media = soma / len(salto)
    print('Média dos saltos: {} m'.format(str(media)))    
salto("Rodrigo", [6.5, 6.1, 6.2, 5.4, 5.3])

Atleta: Rodrigo
Primeiro Salto: 6.5 m
Segundo Salto: 6.1 m
Terceiro Salto: 6.2 m
Quarto Salto: 5.4 m
Quinto Salto: 5.3 m
Resultado final:
Atleta: Rodrigo
Saltos: 6.5 - 6.1 - 6.2 - 5.4 - 5.3
Média dos saltos: 5.9 m


13) Qual é a diferença entre o **while** e **for**?

In [ ]:
while executa enquanto está de acordo com a condição, o for faz o loop até o limite definido.

14) Faça um Programa que peça a idade e a altura de 5 pessoas, armazene cada informação no seu respectivo vetor. Imprima a idade e a altura na ordem inversa a ordem lida.

In [ ]:
def pessoas(idade, altura):
    
pessoas()

15) Faça um Programa que leia dois vetores com 10 elementos cada. Gere um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos intercalados dos dois outros vetores.

16) Faça um programa que receba a temperatura média de cada mês do ano e armazene-as em uma lista. Após isto, calcule a média anual das temperaturas e mostre todas as temperaturas acima da média anual, e em que mês elas ocorreram (mostrar o mês por extenso: 1 – Janeiro, 2 – Fevereiro,...).

NumPy is installed


17) Faça uma função que recebe um valor n e que imprima:
  
    1
  
    2   2
  
    3   3   3
  
    .....
  
    n   n   n   n   n   n  ... n

18) Faça uma função que recebe um valor n e que imprima:
  
  1
  
  1   2
  
  1   2   3
  
  .....
  
  1   2   3   ...  n

19) Faça uma função que retorne o reverso de um número inteiro passado por parâmetro. Por exemplo: 127 -> 721.

20) Faça uma função que receba dois vetores com 10 elementos cada. Sua função deverá retornar um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos intercalados dos dois outros vetores.

21) Faça uma função que receba uma string e retorne a quantidade de consoantes que existem nela.

22) Altere o exercício 21 para que o retorno seja um dicionário, onde a chaves são as consoantes e os valores a quantidade de vezes que a consoante aparece na string

#### **Boa sorte!**